src/main/java/org/example/internetshop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,187 @@
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getSurname());
+            statement.setString(3, user.getLogin());
+            statement.setString(4, user.getPassword());
+            statement.setLong(5, user.getId());
+            statement.executeUpdate();
+            query = "DELETE FROM users WHERE id=?;";
+            clearUserInfo(query, connection, user.getId());
+            setUserRoles(user);
+            ResultSet resultSet = statement.getResultSet();
+            return getUserFromResultSet(resultSet);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Update user with id = "
+                    + user.getId() + "failed", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "DELETE FROM users_roles WHERE user_id=?;";
+            clearUserInfo(query, connection, id);
+            query = "DELETE FROM users WHERE id=?;";
+            return clearUserInfo(query, connection, id);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Deleting user with id = "
+                    + id + " failed", e);
+        }
+    }
+
+    @Override
+    public List<User> getAll() {
+        String query = "SELECT * FROM users u "
+                + "JOIN users_roles ur ON u.id=ur.user_id;";
+        List<User> users = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.executeQuery();
+            ResultSet resultSet = statement.getResultSet();
+            while (resultSet.next()) {
+                users.add(getUserFromResultSet(resultSet));
+            }
+            return users;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting all users failed", e);
+        }
+    }
+
+    private User getUserFromResultSet(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getLong("id");
+        String name = resultSet.getString("name");
+        String surname = resultSet.getString("surname");
+        String login = resultSet.getString("login");
+        String password = resultSet.getString("password");
+
+        return new User(id, name, surname, login, password, getUserRoles(id));
+    }
+
+    private void setUserRoles(User user) {
+        String query = "INSERT INTO users_roles (user_id, role_id)"
+                + "VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            for (Role role : user.getRoles()) {
+                statement.setLong(1, user.getId());
+                statement.setLong(2, role.getId());
```

Comment:
where did you role get id on this stage?

Reply:
i setted them on role model. But, ok, i understood how to get it from db.